roborazzi-idea-plugin/src/main/kotlin/com/github/takahirom/roborazzi/idea/preview/RoborazziPreviewTool.kt

```diff
@@ -86,222 +65,11 @@ class RoborazziPreviewToolWindowFactory : ToolWindowFactory {
-    val classes: List<KtClass> = allDeclarations.filterIsInstance<KtClass>()
-      .filter {
-        it.name?.contains("Test") == true || it.declarations.any {
-          hasPreviewOrTestAnnotationOrHasNameOfTestFunction(it)
-        }
-      }
-
-    val searchPath = project.basePath
-    statusText.value = "Searching images in $searchPath ..."
-
-    val files = withContext(Dispatchers.IO) {
-      val roborazziFolders = ProjectRootManager.getInstance(project).contentRootsFromAllModules
-        .map { File(it.path + AppSettingsState.instance.imagesPathForModule) }
-        .filter {
-          it.exists()
-        }
-
-      roborazziFolders
-        .flatMap { folder ->
-          folder.walkTopDown().filter { it.isFile }
-        }
-    }
-
-    allPreviewImageFiles.addAll(findImages(classes, files))
-    allPreviewImageFiles.addAll(findImages(functions, files))
-
-    if (allPreviewImageFiles.isEmpty()) {
-      statusText.value = "No images found"
-    } else {
-      statusText.value = "${allPreviewImageFiles.size} images found"
-    }
-    val result = allPreviewImageFiles.sortedByClassesAndFunctions(classes, functions)
-      .map { it.path to it.lastModified() }
-    roborazziLog("refreshListProcess result result.size:${result.size} in ${System.currentTimeMillis() - start}ms")
-    imagesStateFlow.value = result
-  }
-
-  private suspend fun List<File>.sortedByClassesAndFunctions(
-    classes: List<KtClass>,
-    functions: List<KtFunction>
-  ): List<File> {
-    val allFunctionNamesOrder = (classes
-      .flatMap { it.declarations.filterIsInstance<KtFunction>() } + functions
-      ).map { it.name }
-    val files = this
-    return withContext(Dispatchers.Default) {
-      files.sortedBy { file ->
-        allFunctionNamesOrder.indexOfFirst { functionName ->
-          file.name.contains(functionName ?: "")
-        }
-      }
-    }
-  }
-
-  private suspend fun findImages(
-    elements: List<KtElement>,
-    files: List<File>
-  ): List<File> {
-    val elementNames = elements.mapNotNull { element ->
-      element.name
-    }
-    return withContext(Dispatchers.Default) {
-      elementNames.flatMap { elementName ->
-        val pattern = ".*$elementName.*.png"
-        files
-          .filter {
-            val matches = it.name.matches(Regex(pattern))
-            matches
-          }
-      }
-    }
-  }
-
-  private fun cancel() {
-    coroutineScope.cancel()
-  }
-
-  fun onHide() {
-    cancel()
-  }
-
-  fun onShouldSeeIndexHandled() {
-    shouldSeeIndex.value = -1
-  }
-}
-
 class RoborazziPreviewPanel(project: Project) : JPanel(BorderLayout()) {
   private val listModel = DefaultListModel<Pair<String, Long>>()
-  private val statusLabel = JBLabel("No images found")
+  private val statusGradleTaskPanel = StatusToolbarPanel(project) { taskName ->
+    viewModel?.executeTaskByName(project, taskName)
+  }
   private val statusBar = JBBox.createHorizontalBox().apply {
     add(JLabel("Refresh: ").apply {
```

Comment:
I think we no longer need this JLabel("Refresh: "). Thanks to the new refresh button 🙏